intermediate_source/torchrec_interactive_tutorial.py

```diff
@@ -0,0 +1,1107 @@
+
+print(f"Type of optimizer: {type(sharded_ebc_fused_params.fused_optimizer)}")
+
+from torch.distributed.optim import _apply_optimizer_in_backward as apply_optimizer_in_backward
+import copy
+# Option 2: Applying optimizer through apply_optimizer_in_backward
+# Note: we need to call apply_optimizer_in_backward on unsharded model first and then shard it
+
+# We can achieve the same result as we did in the previous
+ebc_apply_opt = copy.deepcopy(ebc)
+optimizer_kwargs = {"lr": 0.5}
+
+for name, param in ebc_apply_opt.named_parameters():
+    print(f"{name=}")
+    apply_optimizer_in_backward(torch.optim.SGD, [param], optimizer_kwargs)
+
+sharded_ebc_apply_opt = sharder.shard(ebc_apply_opt, plan.plan[""], env, torch.device("cuda"))
+
+# Now when we print the optimizer, we will see our new learning rate, you can verify momentum through the TBE logs as well if outputted
+print(sharded_ebc_apply_opt.fused_optimizer)
+print(type(sharded_ebc_apply_opt.fused_optimizer))
+
+# We can also check through the filter other parameters that aren't associated with the "fused" optimizer(s)
+# Practically, just non TorchRec module parameters. Since our module is just a TorchRec EBC
+# there are no other parameters that aren't associated with TorchRec
+print("Non Fused Model Parameters:")
+print(dict(in_backward_optimizer_filter(sharded_ebc_fused_params.named_parameters())).keys())
+
+# Here we do a dummy backwards call and see that parameter updates for fused
+# optimizers happen as a result of the backward pass
+
+ebc_output = sharded_ebc_fused_params(kjt).wait().values()
+loss = torch.sum(torch.ones_like(ebc_output) - ebc_output)
+print(f"First Iteration Loss: {loss}")
+
+loss.backward()
+
+ebc_output = sharded_ebc_fused_params(kjt).wait().values()
+loss = torch.sum(torch.ones_like(ebc_output) - ebc_output)
+# We don't call an optimizer.step(), so for the loss to have changed here,
+# that means that the gradients were somehow updated, which is what the
+# fused optimizer automatically handles for us
+print(f"Second Iteration Loss: {loss}")
+
+
+######################################################################
+# Inference
+# ~~~~~~~~~~~~
+# 
+# Now that we are able to train distributed embeddings, how can we take
+# the trained model and optimize it for inference? Inference is typically
+# very sensitive to **performance and size of the model**. Running just
+# the trained model in a Python environment is incredibly inefficient.
+# There are two key differences between inference and training
+# environments: \* **Quantization**: Inference models are typically
+# quantized, where model parameters lose precision for lower latency in
+# predictions and reduced model size. For example FP32 (4 bytes) in
+# trained model to INT8 (1 byte) for each embedding weight. This is also
+# necessary given the vast scale of embedding tables, as we want to use as
+# few devices as possible for inference to minimize latency. \* **C++
```

Comment:
the bulleted lists don't render as such 
<img width="867" alt="image" src="https://github.com/user-attachments/assets/032cef01-cfc6-41ae-8f5d-5347f465f5de">